subsys/nrf_security/Kconfig.legacy

```diff
@@ -76,81 +76,81 @@ config MBEDTLS_AES_DECRYPT_ALT
 
 config MBEDTLS_AES_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_CIPHER_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_CMAC_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_CMAC_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_CCM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_CCM_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_GCM_ALT
 	bool
 	depends on CC312_BACKEND || \
-		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE)
+		   (PSA_CRYPTO_DRIVER_HAS_GCM_SUPPORT_CC3XX && CRYPTOCELL_CC312_USABLE)
 	default y
 
 config MBEDTLS_CHACHA20_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 
 config MBEDTLS_POLY1305_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 
 config MBEDTLS_CHACHAPOLY_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 	depends on MBEDTLS_CHACHA20_ALT
 
 config MBEDTLS_DHM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_RSA_SUPPORT_CC3XX
```

Comment:
DHM is RSA?
From the mbedtls docs this looks to be "diffie hellmann merkle", should this be ECC instead?

Reply:
yes

Reply:
Kconfig changed to new type (ocrypto Kconfig rewrite)